src/hpa.c

```diff
@@ -537,9 +537,15 @@ hpa_shard_maybe_do_deferred_work(tsdn_t *tsdn, hpa_shard_t *shard,
 		purged = false;
 		while (hpa_should_purge(tsdn, shard) && nops < max_ops) {
 			purged = hpa_try_purge(tsdn, shard);
-			if (purged) {
-				nops++;
+			/*
+			 * It is fine if we couldn't purge as sometimes we try
+			 * to purge just to unblock hugification, but there is maybe
+			 * no dirty pages at all at the moment.
+			 */
+			if (!purged) {
+				break;
 			}
+			nops++;
 		}
```

Comment:
nit: maybe a bit more readable this way?
```
if (!hpa_try_purge(tsdn, shard)) {
  // comment
  break;
}
nops++
```

Reply:
We still need to update `purged` as it is used to terminate loop at line 550.

I guess the best we can do is this, but not sure it is more readable to be honest as line seems a bit overloaded: negation, assignment, function call and condition checking it one line. 
```
if (!(purged = hpa_try_purge(tsdn, shard))) {
        // comment;
        break;
}
```
Moved comment inside if though.